Update workflow, script, and submodules

L1ghtmann opened the pull request:
- Updates `libplist`
- Updates `ldid`
- Adjusts `libtapi` build to slim archs and targets built 
- Adds libplist's `plistutil` to toolchain 
- Simplifies parts of workflow and script 
- Switches to utilizing docker for builds 
- Adds aarch64 builds
- Adds native Arch and UBI builds

Would've also updated `cctools-port` and `apple-libtapi`, but they seem to have issues building. 

Tested semi-successfully [here](https://github.com/L1ghtmann/darwin-tools-linux/actions/runs/9339562095). Unsure how to go about resolving the Arch-aarch64 and UBI-aarch64 builds failing due to time constraints...open to changes there.

L1ghtmann committed:
Update release.yml

L1ghtmann committed:
Workflow cleanup

L1ghtmann committed:
Merge branch 'master' into pr

L1ghtmann committed:
Update workflow

L1ghtmann committed:
Update libplist & ldid

L1ghtmann committed:
Simplify and reorder a couple of things for clarity

L1ghtmann committed:
Fix

L1ghtmann committed:
Copy plistutil to staging

L1ghtmann committed:
Fix release name (whoops)

L1ghtmann committed:
I'm tired :/

kabiroberai commented on prepare-toolchain:
```diff
@@ -51,6 +51,7 @@ PLIST_PREFIX="${STAGING}/libplist-install"
 mkdir -p "${PLIST_PREFIX}"
 LC_ALL=C ./autogen.sh --prefix="${PLIST_PREFIX}" --without-cython PACKAGE_VERSION=latest
 make -j "${JOBS}" install
+cp -av "${PLIST_PREFIX}/bin/plistutil" "${INSTALLPREFIX}/bin"
```
is `plistutil` statically linked to libplist? If not we'll either need to install libplist into the prefix (ensuring that the linkage is via `$ORIGIN`) or change it to link statically. In the former case we'll probably want to make `ldid` link against it dynamically as well.

L1ghtmann replied:
Not by default. Appears to be after passing `--enable-static --disable-shared` to autogen.

L1ghtmann committed:
Build libplist statically

L1ghtmann committed:
Switch to docker, add AArch64, Arch, and UBI builds, update libplist

L1ghtmann commented:
Recent updates include:
- Switching to docker for the builds
- Support for Arch and UBI native builds
- AArch64 build support 
- Updates to workflow to use artifacts for incremental builds and a release for the complete final build
- Updates libplist submodule

Unfortunately, the workflow is only [semi-successful](https://github.com/L1ghtmann/darwin-tools-linux/actions/runs/9339562095) due to very slow Arch-aarch64 and UBI-aarch64 builds. Open to suggestions as to how to remedy that.

L1ghtmann committed:
Merge branch 'master' into pr

L1ghtmann committed:
Switch to my tapi fork for time being to fix linking

L1ghtmann committed:
Fix build dir/src dir conflict(?)

L1ghtmann committed:
-_-

L1ghtmann commented:
Pr has been updated. Changes from upstream include:
- Simplification of CI
- Addition of Arch build
- Update to all submodules 
  - Switched (lib)tapi to my fork for now until fix is merged
- Updated build script to simplify a few things and slim the backend for tapi 
- Static libplist + `plistutil` added to toolchain bin

Removed my previously added aarch64 builds since they still fail (Docker takes forever). 

Edit: if interested in this pr, will want to squash as the history is 2-3 years old and gross (apologies)

kabiroberai commented on .github/workflows/release.yml:
```diff
@@ -40,28 +21,40 @@ jobs:
           - name: fedora39
             image: fedora:39
-    runs-on: ubuntu-24.04
+          - runner: archlinux
```
should this be `name` instead of `runner`?
```suggestion
          - name: archlinux
```

L1ghtmann replied:
Oop, nice catch

kabiroberai commented on prepare-toolchain:
```diff
@@ -23,24 +23,22 @@ mkdir -p "${INSTALLPREFIX}"
 	-DCMAKE_BUILD_TYPE=RELEASE \
 	-DCMAKE_INSTALL_PREFIX="$INSTALLPREFIX"
-make -j "${JOBS}" install
+cmake --build libdispatch --target install
```
should we pass `-j` here as well?
```suggestion
cmake --build libdispatch --target install -- -j "${JOBS}"
```

L1ghtmann replied:
yup - apologies for missing that

kabiroberai commented on .gitmodules:
```diff
@@ -1,6 +1,7 @@
 	path = vendor/apple-libtapi
-	url = https://github.com/tpoechtrager/apple-libtapi.git
+	url = https://github.com/l1ghtmann/apple-libtapi.git
+	branch = tapi-link-fix
```
looks like your PR was just merged(!) let's update this back to upstream?

kabiroberai reviewed (commented):
LGTM overall with a few nits/questions. Thanks again!

L1ghtmann committed:
Requested changes
